common/common.cpp

```diff
@@ -156,7 +421,168 @@ static int count_math_cpus(int cpu_count) {
     return result;
 }
 
-#endif // __x86_64__ && __linux__
+#elif defined(_WIN32)
+
+#define STATUS_ACCESS_DENIED ((NTSTATUS)0xC0000022L)
+#define STATUS_SUCCESS		 ((NTSTATUS)0)
+
+typedef enum _SYSTEM_INFORMATION_CLASS {
+	SystemAllowedCpuSetsInformation = 168,
+	SystemCpuSetInformation = 175,
+	SystemCpuSetTagInformation = 176,
+} SYSTEM_INFORMATION_CLASS;
+
+typedef enum _PROCESSINFOCLASS {
+	ProcessDefaultCpuSetsInformation = 66,
+	ProcessAllowedCpuSetsInformation = 67,
+} PROCESSINFOCLASS;
+
+extern "C"
+NTSTATUS
+NTAPI
+NtQuerySystemInformationEx(
+	_In_ SYSTEM_INFORMATION_CLASS SystemInformationClass,
+	_In_reads_bytes_(InputBufferLength) PVOID InputBuffer,
+	_In_ ULONG InputBufferLength,
+	_Out_writes_bytes_opt_(SystemInformationLength) PVOID SystemInformation,
+	_In_ ULONG SystemInformationLength,
+	_Out_opt_ PULONG ReturnLength
+);
+
+
+extern "C"
+NTSTATUS
+NTAPI
+NtQueryInformationProcess(
+	_In_ HANDLE ProcessHandle,
+	_In_ PROCESSINFOCLASS ProcessInformationClass,
+	_Out_writes_bytes_opt_(ProcessInformationLength) PVOID ProcessInformation,
+	_In_ ULONG ProcessInformationLength,
+	_Out_opt_ PULONG ReturnLength
+);
```

Comment:
These forward declarations don't seem to be used, so you should remove them.

Reply:
> These forward declarations don't seem to be used, so you should remove them.

Thanks for noticing it, I was using it for something else that I removed later.

Right now I'm adding support for Linux.
Found out the implementation was bugged; there is a typo in the sys path and the affinity for the process is never set.
I wouldn't be able to support the same as Windows but except the last level cache traversal everything else should be fine.
Got a nice 10% t/s speed up with a 5600G on Debian.